nle/nethack/nethack.py

```diff
@@ -91,19 +99,24 @@ def _new_dl_linux(vardir):
     dl = tempfile.TemporaryFile(suffix="libnethack.so", dir=vardir)
     path = "/proc/self/fd/%i" % dl.fileno()
     shutil.copyfile(DLPATH, path)  # Should use sendfile.
+
     return dl, path
 
 
 def _new_dl(vardir):
     """Creates a copied .so file to allow for multiple independent NLE instances"""
-    if sys.platform == "linux":
+
+    # Linux has memfd_create or O_TMPFILE,
+    # but both methods are not compatible with fixing rpath with patchelf.
+    if sys.platform == "linux" and not os.path.exists(PACKAGE_LIBS_DIR):
```

Comment:
This means this method depends on having to physically copy the .so file?

Could you explain what patchelf does here? It sounds like you need to adjust the rpath of the .so. Could we do that once and afterwards still "copy" the file in the cheap way? I'd much prefer not running into the danger of creating many copies of this file in cases where NLE is closed uncleanly.

Reply:
Yes, you are right, patchelf fixes rpath after coping libnethack.so from site-packages/nle. I was testing it before directly on /proc/self/fd/ files but patchelf was failing. I tested it again, and it seems to work, so I had to make some mistake previously. I pushed a commit, with the change.

Reply:
Tested both memfd_create as well as 0_TMPFILE.